src/app/Library/CrudPanel/Traits/Update.php

```diff
@@ -113,69 +115,69 @@ private function getModelAttributeValueFromRelationship($model, $field)
-                    return $field['name'] != $item['name'];
-                });
 
                 $related_models = $related_model->{$relation_method};
                 $result = [];
 
-                // for any given model, we grab the attributes that belong to our pivot table.
                 foreach ($related_models as $related_model) {
                     $item = [];
                     switch ($relation_type) {
                         case 'HasMany':
                         case 'MorphMany':
-                            $result[] = $related_model->getAttributes();
+                            // we will get model direct attributes and merge with subfields values.
+                            $directAttributes = $related_model->withFakes()->getAttributes();
+                            $result[] = array_merge($directAttributes, $this->getSubfieldsValues($field['subfields'], $related_model));
                             break;
 
                         case 'BelongsToMany':
                         case 'MorphToMany':
+                            // for any given model, we grab the attributes that belong to our pivot table.
                             $item = $related_model->pivot->getAttributes();
-                            $item[$field['name']] = $related_model->getKey();
+                            $item[$relation_method] = $related_model->getKey();
                             $result[] = $item;
                             break;
                     }
                 }
 
                 return $result;
-
                 break;
             case 'HasOne':
             case 'MorphOne':
                 if (! method_exists($related_model, $relation_method)) {
                     return;
                 }
 
-                $related_entry = $related_model->{$relation_method};
+                $related_entry = $related_model->{$relation_method}->withFakes();
 
                 if (! $related_entry) {
                     return;
                 }
 
+                // if `entity` contains a dot here it means developer added a main HasOne/MorphOne relation with dot notation
                 if (Str::contains($field['entity'], '.')) {
                     return $related_entry->{Str::afterLast($field['entity'], '.')};
                 }
 
+                // when subfields exists developer used the repeatable interface to manage this relation
                 if ($field['subfields']) {
-                    $result = [];
-                    foreach ($field['subfields'] as $subfield) {
-                        $name = is_string($subfield) ? $subfield : $subfield['name'];
-                        $result[$name] = $related_entry->{$name};
-                    }
-
-                    return [$result];
+                    return [$this->getSubfieldsValues($field['subfields'], $related_entry)];
                 }
 
-                return $related_entry;
+                return $related_entry->withFakes();
```

Comment:
Same here... what if the related entry doesn't have a CRUD Panel? Say... it's from a package... like one from Spatie or something.